OnionUI/Onion
# Add RetroArch local hotspot quick host/join

## Overview
- Takes all the hassle out of launching games with your friends locally.
- Quick one-tap feature to host a netplay capable game and core over the hotspot.
- Quick one-tap feature to find the hotspot on another (single atm as retro joins as P2 always) MMP and sync the core/rom with the host, then join.

## Features

- Checksum comparisons to cut down on unnecessary downloads and hopefully have a "get it right first time" approach to launching RA with a matching core & rom for higher compatibility.
- Adds infopanels to host script
- Adds the client side script with infopanels
- Manages WIFI and tries to reconnect WIFI back to the original network after host/client exit the game for better user experience

## TODO 

- Needs fully testing on other devices.
- Currently doesn't support contentless cores. (mr.boom etc)

## Test process:

- Set your default core in GLO as a netplay capable core and use a netplay capable game. https://docs.google.com/spreadsheets/d/1y1esG3HcNagPG8cy-ulZK_mhhUb93pqG5_XEtRRgKQ4/edit#gid=0
- On MMP1 start the hotspot through GLO on the game you want to play (Netplay - Host game on hotspot)
- Wait for RA to load fully in on MMP1
- On MMP2 click Netplay - Join local host (can be a script in GLO, an app or called from anywhere really)
- If WIFI is disabled it will be brought up.
- If there's a core/rom mismatch this will be fixed
- MMP2 will now configure itself and join the RA server as player 2.
- Play game
- Exit RA on the host, observe WIFI is restored
- Exit RA on the client, observe WIFI is restored
- Fin

## Notes/Other changes
- Added some log removal to runtime.sh
- wlan0 will be set `down` on MMP1 while the hotspot is running to improve wifi performance
- The client will sync the core/rom but the old core/rom will be renamed with `_old` as to not wipe the users original cores/roms for the user. These are not automatically restored yet.
- Doesn't currently add the App, both methods can be used through GLO on any game/location

## Issues
- DHCP can sometimes take a while when wifi is turned off.

## Video examples
### Host
https://github.com/OnionUI/Onion/assets/47260768/1b98e695-6124-4ff5-bd2a-7fee87d0ab58


### Client
https://github.com/OnionUI/Onion/assets/47260768/8533d76e-92e0-45e8-9b9a-85667c8b363d

## Screenshots
![Screenshot_007](https://github.com/OnionUI/Onion/assets/47260768/f82817ad-90d2-4a60-97ab-13d1034cb29f)
